jekyll/_cci2/deployment-examples.md

```diff
@@ -0,0 +1,678 @@
+          image: <your-image> # name of your Docker image
+          tag: $CIRCLE_SHA1 # Docker image tag - optional
+```
+### Deployment to GKE with 2.0 Config
+
+In the following example, if the `build-job` passes and the current branch is `master`, CircleCI runs the deployment job.
+
+{% raw %}
+```
+version: 2
+
+jobs:
+ build-job:
+  # steps ommitted for brevity
+ deploy-job:
+   docker:
+     - image: my-image-version-tag
+   working_directory: /tmp/my-project  
+   steps:
+     - run:
+         name: Deploy Master to GKE
+         command: |
+         # Push Docker image to registry, update K8s deployment to use new image - `gcloud` command handles authentication and push all at once
+         sudo /opt/google-cloud-sdk/bin/gcloud docker push us.gcr.io/${PROJECT_NAME}/hello 
+         # The new image is now available in GCR for the GCP infrastructure to access, next, change permissions:
+         sudo chown -R ubuntu:ubuntu /home/ubuntu/.kube
+         # Use `kubectl` to find the line that specifies the image to use for our container, replace with image tag of the new image. 
+         # The K8s deployment intelligently upgrades the cluster by shutting down old containers and starting up-to-date ones.
+         kubectl patch deployment docker-hello-google -p '{"spec":{"template":{"spec":{"containers":[{"name":"docker-hello-google","image":"us.gcr.io/circle-ctl-test/hello:'"$CIRCLE_SHA1"'"}]}}}}'
+
+workflows:
+  version: 2
+  build-deploy:
+    jobs:
+      - build-job 
+      - deploy-job:
+          requires:
+            - build-job # Only deploy once the build job has completed
+          filters:
+            branches:
+              only: master # Only deploy on the master branch
+
+```
+{% endraw %}
+
+For another example, see our [CircleCI Google Cloud deployment example project](https://github.com/CircleCI-Public/circleci-demo-k8s-gcp-hello-app).
+
+## Heroku
+
+[Heroku](https://www.heroku.com/) is a popular platform for hosting applications in the cloud. To configure CircleCI
+to deploy your application to Heroku, follow the steps below.
+
+### Deploy with the Heroku Orb
+
+1. Create a Heroku account and follow the [Getting Started on Heroku](https://devcenter.heroku.com/start) documentation
+to set up a project in your chosen language.
+
+2. Add the name of your Heroku application and your Heroku API key as environment variables as `HEROKU_APP_NAME` and `HEROKU_API_KEY`, respectively. {% include snippets/env-var-or-context.md %}
+
+3. Use the [Heroku orb](https://circleci.com/orbs/registry/orb/circleci/heroku) to keep your config simple. The `deploy-via-git` installs the Heroku CLI in the primary container, runs any pre deployment steps you define, deploys your application, then runs any post-deployment steps you define. See the Heroku orb page in the [orbs registry](https://circleci.com/orbs/registry/orb/circleci/heroku) for full details of parameters and options:
```

Comment:
do the pre-deploy and post-deploy steps get added within the workflows sections of the config? There isn 't a usage example for these bits in the orb ref page.